doc/signal_path.rst

```diff
@@ -0,0 +1,302 @@
+Signal path overview
+====================
+
+This section gives a logical overview of the signal path. The actual
+implementation is mathematically equivalent (when ignoring floating-point
+rounding errors), but it splits, combines, and reorders steps for efficiency;
+refer to the :doc:`fgpu.design` and :doc:`xbgpu.design` sections for details.
+
+Edges in diagrams are annotated to indicate the data type. The following types
+are used:
+
+:samp:`i{N}`
+  Signed integer or fixed-point with :samp:`{N}` total bits
+:samp:`f{N}`
+  Floating-point with :samp:`{N}` bits, following IEEE 754-2019.
+:samp:`c{X}`
+  Complex values formed from real and imaginary components of type :samp:`{X}`
+  e.g., ``cf32`` is single-precision floating point complex.
+
+Dotted boxes and arrows represent control parameters that can be adjusted at
+runtime.
+
+Channelisation
+--------------
+Note that the input and output bit depths (shown as ``i10`` and ``ci8`` on the
+diagram) are configurable. Between unpacking and quantisation, all
+calculations are performed in single precision. Since the input has a bounded
+range, overflow is only possible at the quantisation step (which saturates).
+
+The figure below shows the signal path for wide-band channelisation.
+
+.. tikz:: Signal path for wide-band channelisation.
+   :libs: chains, positioning
+
+   \tikzset{
+     base/.style={minimum width=2.5cm, minimum height=1cm, align=center},
+     op/.style={draw, base},
+     control/.style={draw, base, rounded corners, dotted},
+     lbl/.style={font=\scriptsize},
+     every join/.style={draw,->},
+     >=latex,
+   }
+   \newcommand{\side}[2]{
+     \node[op, on chain, join=by {#2, edge label=f32}] (cdelay#1) {Coarse delay};
+     \node[op, on chain, join=by {#2, edge label=f32}] (pfb#1) {PFB};
+     \node[op, on chain, join=by {#2, edge label=cf32}] (eq#1) {Fine delay\\ Equalisation};
+     \node[op, on chain, join=by {#2, edge label=cf32}] (dither#1) {Dither};
```

Comment:
Obviously noting that we may have to remove this step in the future if SE decides it's ugly.